pageserver/src/tenant/storage_layer/image_layer.rs

```diff
@@ -544,7 +544,7 @@ impl ImageLayerInner {
             let mut search_key: [u8; KEY_SIZE] = [0u8; KEY_SIZE];
             range.start.write_to_byte_slice(&mut search_key);
 
-            let index_stream = tree_reader.get_stream_from(&search_key, &ctx);
+            let index_stream = tree_reader.clone().into_stream(&search_key, &ctx);
```

Comment:
The `.clone()` and the `#[derive(Clone)]`s are unnecessary if you simply construct the `block_reader` and `tree_reader` in the body o fthe loop.

Reply:
There is another case where the reader is passed as a parameter so I decided to clone for here and that place.